content/blog/bare-metal-provisioning-on-hpe-greenlake-using-terraform.md

```diff
@@ -0,0 +1,359 @@
+* machine_size – Compute Instance type.
+* ssh – List of SSH keys to push to the host.
+* location – The data center location of where the compute instance will be provisioned.
+* Image – OS image in the form flavor@version.
+
+  Y﻿ou can also check the documentation [here](https://registry.terraform.io/providers/HPE/hpegl/latest/docs/resources/metal_host) to see all the required and optional fields.
+
+  Your next step with the TF file is to query the HPE GreenLake provider to collect the above-required information for creating a host. For this, you will use the Terraform data statements.  
+
+### Querying for available OS images
+
+In order to list the available OS images for OS, add the below data statements in your terraform file ***main.tf***:
+
+```hcl
+data "hpegl_metal_available_images" "ubuntu" { 
+  # select anything that looks like ubuntu:20.04
+  filter {
+    name   = "flavor"
+    values = ["(?i)ubuntu"] 
+  }
+
+  filter {
+    name   = "version"
+    values = ["20.04*"] 
+  }
+}
+```
+
+ The OS image list can be fetched using the following statements in main.tf: 
+
+```hcl
+locals {
+  ubuntu_image = format("%s@%s", data.hpegl_metal_available_images.ubuntu.images[0].flavor,
+                          data.hpegl_metal_available_images.ubuntu.images[0].version)
+}
+```
+
+### Querying for other available resources
+
+For this, you should use **hpegl_metal_available_resources** data resource. For example, the following statements show how to retrieve and store the available SSH keys in a local variable.\
+\
+Append to main.tf:
+
+```hcl
+# query available resources.
+data "hpegl_metal_available_resources" "available" {
+}
+
+# using one of the available SSH keys.
+locals  {
+  ssh_keys = data.hpegl_metal_available_resources.available.ssh_keys[0].name
+}
+```
+
+Using a similar technique, you can retrieve the rest of the data you need - networks, machine size, etc.\
+\
+A﻿ppend to main.tf:
+
+```hcl
+# choosing a location that has at least one machine available.
+locals {
+  location = ([for msize in data.hpegl_metal_available_resources.available.machine_sizes : msize.location 
+                    if msize.quantity > 0])[0]
+}
+
+# Listing required networks for this location.
+locals {
+  networks = ([for net in data.hpegl_metal_available_resources.available.networks : net.name 
+                  if net.host_use == "Required"  && net.location == local.location])
+}
+
+# choosing machine size/Compute Instance Type to deploy OS.
+locals {
+  machine_size = ([for msize in data.hpegl_metal_available_resources.available.machine_sizes : msize.name 
+                    if msize.location == local.location])
+}
+```
+
+\
+[Here ](https://registry.terraform.io/providers/HPE/hpegl/latest/docs/data-sources/metal_available_resources)you can get information about each of the bare metal data statements supported by the **hpegl** provider.
```

Comment:
It's odd, though, because I don't think the backslash is supposed to be at the beginning of this sentence. 
![image](https://user-images.githubusercontent.com/77496261/222976182-dcf9618b-e1e3-4aad-9064-d57338b6554b.png)